### PR TITLE
Fix issue 16697 - Accept __vector as type specialization in is()

### PR DESCRIPTION
**Pull request by LemonBoy**
Ping @WalterBright, is this OK?

### CODE REVIEW

**Comment by WalterBright** [comment]
Yes, but it needs 100% coverage, and also tests to determine the 'deduced' is correct.

**Comment by LemonBoy** [comment]
Got 100% coverage and added some more tests, I hope it's ok now.

File: test/compilable/b16697.d [hunk 1]
```diff
@@ -0,0 +1,12 @@
+version(D_SIMD)
+{
+    static assert(!is(         float       == __vector)); 
+    static assert(!is(         float[1]    == __vector)); 
+    static assert(!is(         float[4]    == __vector)); 
+    static assert( is(__vector(float[4])   == __vector)); 
+    static assert(!is(__vector(float[3])   == __vector)); 
+    static assert(!is(__vector(float[5])   == __vector)); 
+    static assert( is(__vector(float[4]) X == __vector) &&
```
Comment (WalterBright) [inline test/compilable/b16697.d]: This is incorrect. See clause 6 under https://dlang.org/spec/expression.html#IsExpression
```
is(__vector(float[4]) X == __vector)
```
should set `X` to `float[4]`. This enables "peeling" off a layer of the type.

Comment (LemonBoy) [inline test/compilable/b16697.d]: Fixed!

Comment (WalterBright) [inline test/compilable/b16697.d]: That was fast!

**Pull request merged**